awscli/customizations/s3/fileinfobuilder.py

```diff
@@ -18,13 +20,12 @@ class FileInfoBuilder(object):
     This class takes a ``FileBase`` object's attributes and generates
     a ``FileInfo`` object so that the operation can be performed.
     """
-    def __init__(self, service, endpoint, source_endpoint=None,
+    def __init__(self, client, source_client=None,
                  parameters = None, is_stream=False):
-        self._service = service
-        self._endpoint = endpoint
-        self._source_endpoint = endpoint
-        if source_endpoint:
-            self._source_endpoint = source_endpoint
+        self._client = client
+        self._source_client = copy.copy(client)
```

Comment:
I think I was meaning to do deep copy here. But I do not think a deep copy is needed. We do not register any events in the s3handler so it should be fine to use the same client. It is only in the filegenerator that two seperate clients are required but this takes place after the filegenerator.